### PR TITLE
tests: new test static checker

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This new util helps to check the tests format and order of the keys.
In the future, more checks could be included to make sure tests are readable and consistent.

### CODE REVIEW

File: tests/utils/check-tests.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+def check_dir(directory):
+    if not os.path.isdir(directory):
+        print("Checks failed for directory {}".format(directory))
+        print(' - The path is not a directory')
+        return False
+
+    status = True
+    for file in glob.glob(os.path.join(directory, "**/task.yaml"), recursive=True):
```
Comment (bboozzoo) [inline tests/utils/check-tests.py:88]: I think this could use os.walk, but maybe glob will do just as fine

File: run-checks [hunk 2]
```diff
@@ -267,6 +267,11 @@ if [ "$STATIC" = 1 ]; then
     fi
 
     # FIXME: re-add staticcheck with a matching version for the used go-version
+
+    if [ -z "${SKIP_TESTS_CHECK-}" ] || [ "$SKIP_TESTS_CHECK" = 0 ]; then
+        echo "Checking tests formatting"
+        python3 ./tests/utils/check-tests.py -d ./tests
```
Comment (bboozzoo) [inline run-checks:273]: Please chmod +x the file and then you can drop python3 here.

File: tests/utils/check-tests.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        if not check_task(file):
+            status = False
+
+    return status
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--directory', help="path to the directory to check recursively")
```
Comment (bboozzoo) [inline tests/utils/check-tests.py:97]: required=True ?

File: run-checks (hunk 2 above)
Comment (bboozzoo) [inline run-checks:273]: Also, perhaps the file should be named check-test-format or somesuch

File: tests/utils/check-tests.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    'priority',
+    'warn-timeout',
+    'kill-timeout',
+    'environment',
+    'prepare',
+    'restore',
+    'debug',
+    'execute'
+    ]
+MANDATORY_KEYS = [
```
Comment (bboozzoo) [inline tests/utils/check-tests.py:25]: please run `black` formatter on the file

Comment (sergiocazzolato) [reply tests/utils/check-tests.py:25]: done, thanks for the suggestion

File: run-checks (hunk 2 above)
Comment (sergiocazzolato) [reply run-checks:273]: updated

File: tests/utils/check-tests.py (hunk 3 above)
Comment (sergiocazzolato) [reply tests/utils/check-tests.py:97]: sure, thanks

File: run-checks (hunk 2 above)
Comment (sergiocazzolato) [reply run-checks]: done

File: tests/utils/check-tests.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--directory', help="path to the directory to check recursively")
+    return parser
+
+
+def main():
+    parser = _make_parser()
+    args = parser.parse_args()
+
+    if args.directory:
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/utils/check-tests.py]: Initially I did that, but then I remove the file option to make it simpler, I could add that in a following PR

**Comment by sergiocazzolato** [comment]
@bboozzoo thanks for the review, it is ready for a second round

File: tests/utils/check-tests.py (hunk 1 above)
Comment (sergiocazzolato) [reply tests/utils/check-tests.py]: I tried both and glob seems to be easier

Comment (sergiocazzolato) [reply tests/utils/check-tests.py]: I can add that as an improvement, perhaps I need to research a bit more about how to use os.walk

Comment (bboozzoo) [reply tests/utils/check-tests.py]: No it's fine, glob should be ok

File: tests/unit/go/task.yaml [hunk 6]
```diff
@@ -43,6 +43,7 @@ execute: |
             PATH=$PATH \
             GOPATH=/tmp/static-unit-tests \
             ${skip:-} \
+            SKIP_TESTS_CHECK=1 \
```
Comment (bboozzoo) [inline tests/unit/go/task.yaml:46]: SKIP_TESTS_FORMAT_CHECK

File: run-checks [hunk 7]
```diff
@@ -267,6 +267,11 @@ if [ "$STATIC" = 1 ]; then
     fi
 
     # FIXME: re-add staticcheck with a matching version for the used go-version
+
+    if [ -z "${SKIP_TESTS_CHECK-}" ] || [ "$SKIP_TESTS_CHECK" = 0 ]; then
```
Comment (bboozzoo) [inline run-checks:271]: SKIP_TESTS_FORMAT_CHECK?

File: tests/utils/check-test-format [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                    curr_key, SUPPORTED_KEYS
+                )
+            )
+
+    return findings
+
+
+def check_task(filepath):
+    if not os.path.isfile(filepath):
+        print("Checks failed for task {}".format(filepath))
```
Comment (bboozzoo) [inline tests/utils/check-test-format:67]:
```suggestion
        print("Format checks failed for task {}".format(filepath))
```

File: tests/utils/check-test-format [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            findings.append(
+                "key '{}' not included in the suported keys: {}".format(
+                    curr_key, SUPPORTED_KEYS
+                )
+            )
+
+    return findings
+
+
+def check_task(filepath):
```
Comment (bboozzoo) [inline tests/utils/check-test-format:65]: check_task_format

File: tests/utils/check-test-format [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        return False
+
+    filemap = dict()
+    with open(filepath, "r") as task:
+        filemap = yaml.load(task, Loader=yamlordereddictloader.Loader)
+
+    findings = check_keys_order(filemap.keys())
+    findings.extend(check_mandatory_keys(filemap.keys()))
+    if findings:
+        print("Checks failed for task {}".format(filepath))
```
Comment (bboozzoo) [inline tests/utils/check-test-format:78]:
```suggestion
        print("Format checks failed for task {}".format(filepath))
```

File: tests/utils/check-test-format [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+            status = False
+
+    return status
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-d", "--directory", required=True, help="path to the directory to check recursively"
+    )
```
Comment (bboozzoo) [inline tests/utils/check-test-format:104]: And this should be only other change required to use a positional argument.
```suggestion
    parser.add_argument(
        "directory", help="path to the directory to check recursively"
    )
```

File: run-checks [hunk 12]
```diff
@@ -267,6 +267,11 @@ if [ "$STATIC" = 1 ]; then
...
+        echo "Checking tests formatting"
+        ./tests/utils/check-test-format -d ./tests
```
Comment (bboozzoo) [inline run-checks:273]: Let's switch it to positional now. It's a bit awkward if you are required to pass an option with `-d ..`.

```suggestion
        ./tests/utils/check-test-format ./tests
```

File: tests/utils/check-test-format [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+                findings.append(
+                    "Keys '{}' and '{}' do not follow the desired order: {}".format(
+                        last_key, curr_key, SUPPORTED_KEYS
+                    )
+                )
+
+            last_index = curr_index
+            last_key = curr_key
+
+        except ValueError as err:
```
Comment (bboozzoo) [inline tests/utils/check-test-format:55]: err isn't used so we may as well just drop it
```suggestion
        except ValueError:
```

**Review by bboozzoo — COMMENTED** [review]
Some trivial comments. Otherwise looks very close.

File: tests/utils/check-test-format [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+            findings.append(
+                "key '{}' not included in the suported keys: {}".format(
```
Comment (mardy) [inline tests/utils/check-test-format:57]: s/suported/supported/

Comment (mardy) [reply tests/utils/check-test-format:57]: Though maybe "key X is not among the supported keys: ..." would be clearer.

File: tests/unit/go/task.yaml (hunk 6 above)
Comment (sergiocazzolato) [reply tests/unit/go/task.yaml:46]: done

File: run-checks (hunk 7 above)
Comment (sergiocazzolato) [reply run-checks:271]: done

File: tests/utils/check-test-format (hunk 8 above)
Comment (sergiocazzolato) [reply tests/utils/check-test-format:67]: done

File: tests/utils/check-test-format (hunk 9 above)
Comment (sergiocazzolato) [reply tests/utils/check-test-format:65]: done

File: tests/utils/check-test-format (hunk 10 above)
Comment (sergiocazzolato) [reply tests/utils/check-test-format:78]: done

File: run-checks (hunk 12 above)
Comment (sergiocazzolato) [reply run-checks:273]: done

File: tests/utils/check-test-format (hunk 14 above)
Comment (sergiocazzolato) [reply tests/utils/check-test-format:57]: done, thanks for the suggestion

**Review by bboozzoo — APPROVED** [review]
LGTM, thank you

**Review by mardy — APPROVED** [review]

**Comment by bboozzoo** [comment]
Test failure are unrelated. @mvo5 can you land it?

**Pull request merged**